timemachine/fe/free_energy.py

```diff
@@ -170,28 +170,29 @@ def u_kln_by_component_by_lambda(self) -> NDArray:
 @dataclass
 class Trajectory:
     frames: StoredArrays  # (frame, atom, dim)
-    boxes: NDArray  # (frame, dim, dim)
+    _boxes: List[NDArray]  # (frame, dim, dim)
     final_velocities: NDArray  # (atom, dim)
     final_barostat_volume_scale_factor: Optional[float] = None
 
     def __post_init__(self):
         n_frames = len(self.frames)
         assert n_frames > 0
         n_atoms, n_dims = self.frames[0].shape
-        assert self.boxes.shape == (n_frames, n_dims, n_dims)
+        assert len(self.boxes) == n_frames
+        assert self.boxes[0].shape == (n_dims, n_dims)
         assert self.final_velocities.shape == (n_atoms, n_dims)
 
-    @staticmethod
-    def concatenate(ts: Sequence["Trajectory"]) -> "Trajectory":
-        frames = StoredArrays()
-        for t in ts:
-            frames.extend(t.frames)
-
-        boxes = np.concatenate([t.boxes for t in ts])
-        final_velocities = ts[-1].final_velocities
-        final_barostat_volume_scale_factor = ts[-1].final_barostat_volume_scale_factor
+    def extend(self, other: "Trajectory"):
+        """Concatenate another trajectory to the end of this one"""
+        self.frames.extend(other.frames)
+        self._boxes.extend(other._boxes)
+        self.final_velocities = other.final_velocities
+        self.final_barostat_volume_scale_factor = other.final_barostat_volume_scale_factor
 
-        return Trajectory(frames, boxes, final_velocities, final_barostat_volume_scale_factor)
+    # For backwards compatibility
+    @property
+    def boxes(self) -> NDArray:
+        return np.array(self._boxes)
```

Comment:
This seemed prudent, but may not be necessary. I didn't immediately find downstream code that depends on this being an ndarray.